src/std/net/request.ss

```diff
@@ -45,17 +46,32 @@
    (else
     (error "bad header key" str))))
 
-(def (make-http/1.1-headers headers cookies)
+(def (make-http/1.1-headers headers cookies auth)
   (http-headers-cons! headers
-    (http-headers-cons http/1.1-base-headers
-      (http-headers-cookies (or cookies [])))))
+    (chain http/1.1-base-headers
+      (http-headers-cons <> (http-headers-cookies (or cookies [])))
+      (http-headers-cons <> (http-headers-auth auth)))))
 
 (def http/1.1-base-headers
   '(("User-Agent" . "Mozilla/5.0 (compatible; gerbil/1.0)")
     ("Connection" . "close")
     ("Accept" . "*/*")
     ("Accept-Encoding" . "gzip, deflate, identity")))
 
+(def (http-headers-auth auth)
+  (def (basic-auth-header user password)
+    (let ((credentials (u8vector->base64-string
+                        (with-output-to-u8vector
+                         (lambda ()
+                           (display user)
+                           (display ":")
+                           (display password))))))
+      (cons "Authorization" (string-append "Basic " credentials))))
+  (match auth
+    ([basic: user password] [(basic-auth-header user password)])
+    (#f                     [])
+    (_                      (error "bad auth value" auth))))
```

Comment:
Let's say "unknown auth value" in the error message to convey that we don't support it.

Reply:
done!